File: ccc03j2.java
import java.util.Scanner;

public class ccc03j2 {
	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		int[] array = new int[65000];
		
		for (int i = 0; true; i++) {
			array[i] = input.nextInt();
			if (array[i] == 0) {
				break;
			}
		}
		
		double root;
		for (int user : array) {
			if (user != 0) {
				root = Math.sqrt(user);
				if ((int) root == root) {
					System.out.println("Minimum perimeter is " + 4 * (int) root + " with dimensions " + (int) root + " x " + (int)root);
				} else {
					int upRoot = (int) Math.ceil(root) + 1; // not how multiplication works
					int downRoot = (int) Math.ceil(root) - 1;
					System.out.println("Minimum perimeter is " + (2 * upRoot + 2 * downRoot) + " with dimensions " + upRoot + " x " + downRoot);
				}
			} else {
				break;
			}
		}
	}
}

File: ccc03j2.cpp
#include <iostream>
#include <math.h>

int main() {
	std::cin.sync_with_stdio(0);
	std::cin.tie(0);
	double root;
	int area, intRoot;
	std::cin >> area;
	while (area != 0) {
		root = std::sqrt(area);
		intRoot = (int) root;
		if (intRoot == root) {
			std::cout << "Minimum perimeter is " << 4 * intRoot << " with dimensions " << intRoot << " x " << intRoot << '\n';
		} else {
			bool found = false;
			for (int i = intRoot; i >= 1; i--) {
				for (int j = intRoot; i * j <= area; j++) { 
					if (i * j == area) {
						std::cout << "Minimum perimeter is " << 2 * i + 2 * j << " with dimensions " << i << " x " << j << '\n';
						found = true;
						break;
					}
				}
				if (found) break;
			}
		}
		std::cin >> area;
	}

	return 0;
}